### PR TITLE
[SPARK-36873][BUILD][test-maven] Add provided Guava dependency for network-yarn module

### PR DESCRIPTION
**Pull request by sunchao**
### What changes were proposed in this pull request?

Add provided Guava dependency to `network-yarn` module.

### Why are the changes needed?

In Spark 3.1 and earlier the `network-yarn` module implicitly relies on Guava from `hadoop-client` dependency. This was changed by SPARK-33212 where we moved to shaded Hadoop client which no longer expose the transitive Guava dependency. It stayed fine for a while since we were not using `createDependencyReducedPom` so it picks up the transitive dependency from `spark-network-common` instead. However, things start to break after SPARK-36835 where we restored `createDependencyReducedPom` and now it is no longer able to locate Guava classes:

```
build/mvn test -pl common/network-yarn -Phadoop-3.2 -Phive-thriftserver -Pkinesis-asl -Pkubernetes -Pmesos -Pnetlib-lgpl -Pscala-2.12 -Pspark-ganglia-lgpl -Pyarn
...
[INFO] Compiling 1 Java source to /Users/sunchao/git/spark/common/network-yarn/target/scala-2.12/classes ...
[WARNING] [Warn] : bootstrap class path not set in conjunction with -source 8
[ERROR] [Error] /Users/sunchao/git/spark/common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java:32: package com.google.common.annotations does not exist
[ERROR] [Error] /Users/sunchao/git/spark/common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java:33: package com.google.common.base does not exist
[ERROR] [Error] /Users/sunchao/git/spark/common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java:34: package com.google.common.collect does not exist
[ERROR] [Error] /Users/sunchao/git/spark/common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java:118: cannot find symbol
  symbol:   class VisibleForTesting
  location: class org.apache.spark.network.yarn.YarnShuffleService
```

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Tested with the above `mvn` command and it's now passing.

### CODE REVIEW

**Comment by sunchao** [comment]
cc @gengliangwang @srowen @JoshRosen @dongjoon-hyun @HyukjinKwon

**Review by gengliangwang — APPROVED** [review]

**Comment by sunchao** [comment]
Not sure why this was not caught by either the Jenkins or GitHub CI before - maybe because both of them are using SBT instead of Maven? I wonder if the `test-maven` option still works for the Jenkins CI.

File: common/network-yarn/pom.xml [hunk 1]
```diff
@@ -73,6 +73,10 @@
       <artifactId>${hadoop-client-runtime.artifact}</artifactId>
       <version>${hadoop.version}</version>
     </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
```
Comment (viirya) [inline common/network-yarn/pom.xml:78]: The default is `provided`?

**Comment by gengliangwang** [comment]
> Not sure why this was not caught by either the Jenkins or GitHub CI before - maybe because both of them are using SBT instead of Maven? I wonder if the test-maven option still works for the Jenkins CI.

cc @HyukjinKwon could you check the current GA test?

File: common/network-yarn/pom.xml (hunk 1 above)
Comment (cloud-fan) [reply common/network-yarn/pom.xml:78]: I think it's always better to be explicit.

Comment (gengliangwang) [reply common/network-yarn/pom.xml:78]: +1 with @cloud-fan

**Comment by HyukjinKwon** [comment]
Yeah, i think because it uses SBT. Problem is that we can't test all of the combinations. We will have SBT / Maven / Hadoop / JDK / Hive / Python / ... combinations.

**Comment by HyukjinKwon** [comment]
SBT is being used in GA because in order to partially run the tests quickly (so we can run them in parallel). Maven case, we run the basic build and tests in Jenkins but I don't think we test all of these profiles, yes.

File: common/network-yarn/pom.xml (hunk 1 above)
Comment (sunchao) [reply common/network-yarn/pom.xml:78]: Thanks. Added provided.

**Review by cloud-fan — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by gengliangwang** [comment]
I verified the fix locally.
Merging to master/3.2

**Comment by dongjoon-hyun** [comment]
+1, LGTM.

**Pull request closed without merging**